sympy/logic/boolalg.py

```diff
@@ -1897,6 +1903,20 @@ def SOPform(variables, minterms, dontcares=None):
     >>> SOPform([w, x, y, z], minterms, dontcares)
     (y & z) | (z & ~w)
 
+    The terms can also be represented as integers:
```

Comment:
Is this as useful as allowing dict?
```python
for i, m in minterms:
    if isinstance(m, (dict, Dict)):
        d = defaultdict(int)
        d.update(m)
        minterms[i] = [d[v] for v in variables]
...
```

Reply:
With this, you could pass `{v:1,w:1}`.

Reply:
I'm not sure if I follow the dict idea. Passing `{v:1,w:1}` means that all (min)terms with v = 1 and w = 1 are included?

Could indeed make sense. I suggested in #16137 to be able to pass '-' but this would do it at least as well.

I think passing integers is useful, maybe partly because it the format I use to generate LaTeX K-maps etc for teaching. Also, it is much more compact compared to the list format.

Reply:
> Also, it is much more compact compared to the list format.

Agree and I suppose you get good at knowing what to enter for small ints.

The idea with the dicts is that you would send, in the list, dicts saying what the non-zero terms are: `minterms = [{v:1, w:1}, {x:1, y:1}]`

Reply:
The code I suggested would go in the "`_input_to_binlist`" function.

Reply:
I've added this, but with a slightly different interpretation. Now `{v:1, w:1}` means that v = 1 and w = 1 while x and y takes on all other possible values. I think this makes more sense from a user perspective.

It may be that the code can be more efficiently written though.

"Good" thing is that I found a bug in the implementation, probably related to #14700 , so I added a test for that case. Currently with XFAIL.